network/p2p/libp2pConnector.go

```diff
@@ -18,30 +19,47 @@ import (
 type libp2pConnector struct {
 	backoffConnector *discovery.BackoffConnector
 	host             host.Host
+	log              zerolog.Logger
 }
 
 var _ Connector = &libp2pConnector{}
 
-func newLibp2pConnector(host host.Host) (*libp2pConnector, error) {
+func newLibp2pConnector(host host.Host, log zerolog.Logger) (*libp2pConnector, error) {
 	connector, err := defaultLibp2pBackoffConnector(host)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create libP2P connector: %w", err)
 	}
 	return &libp2pConnector{
 		backoffConnector: connector,
 		host:             host,
+		log:              log,
 	}, nil
 }
 
-func (l *libp2pConnector) ConnectPeers(ctx context.Context, ids flow.IdentityList) map[flow.Identifier]error {
+// UpdatePeers is the implementation of the Connector.UpdatePeers function. It connects to all of the ids and
+// disconnects from any other connection that the libp2p node might have.
+func (l *libp2pConnector) UpdatePeers(ctx context.Context, ids flow.IdentityList) map[flow.Identifier]error {
```

Comment:
`ConnectPeers` is now called `UpdatePeers` and it does both creation and removal of connections.